Javatrix.java

```diff
@@ -477,5 +477,37 @@ public Javatrix times(double scalar)
         }
         return actual;
     }
+
+    /**
+     * Get a submatrix.
+     *
+     * @param r - an array of row indices
+     * @param c - an array of column indices
+     * @return submatrix(r(:), c(:))
+     * @throws ArrayIndexOutOfBoundsException if submatrix
+     * has larger dimensions than this.matrix.
+     */
+    public Javatrix  getMatrix(int[] r, int[] c)
+        throws ArrayIndexOutOfBoundsException
+    {
+        m = r.length;
+        n = c.length;
+        int row;
+        int col;
+        if (m > this.getRowDimension() || n > this.getColumnDimension())
+        {
+            String ex = "Submatrix must have dimensions "
+                + "less than or equal to matrix.";
+            throw new ArrayIndexOutOfBoundsException(ex);
+        }
+        Javatrix subMatrix = new Javatrix(m, n);;
+        for (int i = 0; i < m; i++)
+        {
+            row = r[i];
+            col = c[i];
+            subMatrix.matrix[row][col] = this.get(row, col);
```

Comment:
For consistency, I would suggest using the get method instead of accessing the Javatrix matrix field directly (like using the get method).